### ISSUE TITLE
Reduce GC exposure

### ISSUE THREAD

**Issue by annevk** [w3c/ServiceWorker#1127]
According to @wanderview there's a couple places where service workers end up exposing GC, whereas elsehwere in the platform we've made every effort to avoid that.

I'd like to use this issue to ask @wanderview to explain it in more detail and then hopefully together find ways to preserve the GC invariants others have fought so hard for to keep.

cc @smaug---- @domenic

**Comment by jakearchibald** [comment]
@wanderview any details here?

**Comment by wanderview** [comment]
Not really.  I think I was mostly talking about canary style GC detection where you call `postMessage()` in a loop and the receiving side can tell when the messages stop coming.  I can't remember the full context of this issue.

I also didn't really think we needed to fix anything.  IMO hiding GC behavior is a best effort issue, but there are probably always going to be some side channel timing leaks.

**Comment by yoshisatoyanagisawa** [comment]
Bug Triage: I think the issue is stale.
